langchain/src/vectorstores/tests/vectara.int.test.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-process-env */
+import { test, expect } from "@jest/globals";
+import { VectaraStore } from "../vectara.js";
+import { OpenAIEmbeddings } from "../../embeddings/openai.js";
+import { Document } from "../../document.js";
+import { v4 } from "uuid";
+
+test.skip("Vectara Add Documents", async () => {
+  const store = new VectaraStore(new OpenAIEmbeddings(), {
+    customer_id: Number(process.env.VECTARA_CUSTOMER_ID) || 0,
+    corpus_id: Number(process.env.VECTARA_CORPUS_ID) || 0,
+    api_key: process.env.VECTARA_API_KEY || ""
+  }
+  );
+
+  const documents = [
+    new Document({
+      pageContent: "This is a sample text from langchain unit test",
+      metadata: {
+        document_id: v4(), // Generate a random document id
```

Comment:
Can we base the doc_id on the actual document content, i.e. do some kind of MD5 hashing, instead of a random ID? This way, if you try to index the same document twice it would know that document is already there and not duplicate the content.

Reply:
Good idea. Let me see how to do this in JS.

Reply:
@aamirbutt - I see the implementation in the test. But that puts the responsibility for doc_id on the user. Perhaps we want to move that from the metadata to something where that doc_id is computed using hashcode() inside the vectara.ts code itself, so that the user does not have to do this? This is how it's implemented in the python version (https://github.com/hwchase17/langchain/blob/8a44c879c61fe73ec81000d1dfaab6923ebee8b3/langchain/vectorstores/vectara.py#L123)

Reply:
I had a similar thought. Pinecone supports both [user-provided IDs](https://github.com/vectara/langchainjs/blob/main/langchain/src/vectorstores/tests/pinecone.int.test.ts#L27) and [auto-generated IDs](https://github.com/vectara/langchainjs/blob/main/langchain/src/vectorstores/tests/pinecone.int.test.ts#L41). Could we do the same?

Either way, I don't think this is a blocker. We can ship the initial version supporting only user-generated IDs and then add support for auto-generated IDs.